validator/client/validator.go

```diff
@@ -97,53 +129,76 @@ func (v *validator) SlotDeadline(slot uint64) time.Time {
 // beginning of a new epoch.
 func (v *validator) UpdateAssignments(ctx context.Context, slot uint64) error {
 	// Testing run time for fetching every slot. This is not meant for production!
-	if slot%params.BeaconConfig().SlotsPerEpoch != 0 && v.assignment != nil && false {
+	if slot%params.BeaconConfig().SlotsPerEpoch != 0 && v.assignments != nil {
 		// Do nothing if not epoch start AND assignments already exist.
 		return nil
 	}
 
 	ctx, span := trace.StartSpan(ctx, "validator.UpdateAssignments")
 	defer span.End()
-
+	pks := [][]byte{}
+	for i := range v.keys {
+		pks = append(pks, v.keys[i].PublicKey.Marshal())
+	}
 	req := &pb.CommitteeAssignmentsRequest{
 		EpochStart: slot,
-		PublicKey:  [][]byte{v.key.PublicKey.Marshal()},
+		PublicKey:  pks,
 	}
 
 	resp, err := v.validatorClient.CommitteeAssignment(ctx, req)
 	if err != nil {
-		v.assignment = nil // Clear assignments so we know to retry the request.
+		v.assignments = nil // Clear assignments so we know to retry the request.
 		return err
 	}
 
-	v.assignment = resp
+	v.assignments = resp
+	for _, assignment := range v.assignments.Assignment {
+		var proposerSlot uint64
+		var attesterSlot uint64
+		if assignment.IsProposer && len(assignment.Committee) == 1 {
```

Comment:
this is no longer relevant, please check the latest `validator/client/validator.go` for reference

Reply:
done

Reply:
This is still wrong. Please fix. 

A validator should always attest, whether or not they are a proposer.

Reply:
fixed. Thanks